codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -166,18 +169,65 @@ private void constructHandler() {
      * Adds middleware to the operation middleware stack.
      */
     private void populateOperationMiddlewareStack() {
+
+        // Populate middleware's from runtime client plugins
         runtimeClientPlugins.forEach(runtimeClientPlugin -> {
             if (!runtimeClientPlugin.matchesService(model, service)
                     && !runtimeClientPlugin.matchesOperation(model, service, operation)) {
                 return;
             }
 
-            if (runtimeClientPlugin.buildMiddlewareStack().isPresent()) {
-                runtimeClientPlugin.buildMiddlewareStack().get().applyMiddleware(
-                        writer, service, operation, protocolGenerator, "stack"
-                );
+            if (runtimeClientPlugin.registerMiddleware().isPresent()) {
+                MiddlewareRegistrar middlewareRegistrar = runtimeClientPlugin.registerMiddleware().get();
+                writer.writeInline("$T(stack", middlewareRegistrar.getResolvedFunction());
+
+                Symbol functionArgument = middlewareRegistrar.getFunctionArgument();
+                if (functionArgument != null) {
+                        writer.writeInline(", $P", functionArgument);
+                }
```

Comment:
Is `$P` flexable enough to know not to create pointer references for symbols that weren't declared as pointable?

Reply:
As I understand, $P will only create a pointer if you assign a pointable symbol. Using $P is helpful as it works for both value and pointable symbol